### PR TITLE
finished intent test for CreateAccountActivity

### PR DESCRIPTION
**Pull request by Meilin-Lyu**

### CODE REVIEW

File: app/src/androidTest/java/com/cmput301f20t21/bookfriends/ui/login/CreateAccountActivityTest.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+        ViewInteraction confirmPasswordField = onView(
+                allOf(withId(R.id.confirm_password_field), isDisplayed()));
+        confirmPasswordField.perform(replaceText("incorrect Password"), closeSoftKeyboard());
+
+        ViewInteraction emailField = onView(
+                allOf(withId(R.id.email_field), isDisplayed()));
+        emailField.perform(replaceText("testgmail"), closeSoftKeyboard());
+        // needed for async operations (i.e. authentication)
+
```
Comment (zhpeng811) [inline app/src/androidTest/java/com/cmput301f20t21/bookfriends/ui/login/CreateAccountActivityTest.java:53]: remove comments and spaces here

**Review by zhpeng811 — APPROVED** [review]

File: app/src/main/java/com/cmput301f20t21/bookfriends/ui/login/CreateAccountViewModel.java [hunk 2]
```diff
@@ -19,10 +18,20 @@
 
 
 public class CreateAccountViewModel extends ViewModel {
-    private final IAuthRepository authRepository = AuthRepository.getInstance();
-    private final IUserRepository userRepository = UserRepository.getInstance();
+    private IAuthRepository authRepository = AuthRepository.getInstance();
+    private IUserRepository userRepository = UserRepository.getInstance();
```
Comment (QuocTrungTran) [inline app/src/main/java/com/cmput301f20t21/bookfriends/ui/login/CreateAccountViewModel.java:22]: I don't think we need to initialize here, as we have constructors below?

**Review by QuocTrungTran — APPROVED** [review]
Just 1 comment, otherwise it looks good to me :)

File: app/src/main/java/com/cmput301f20t21/bookfriends/ui/login/CreateAccountViewModel.java (hunk 2 above)
Comment (c25vdw) [reply app/src/main/java/com/cmput301f20t21/bookfriends/ui/login/CreateAccountViewModel.java:22]: we can remove the ` = xxxRepository.getInstance()`, but keep the `final` keyword as we want to be sure they are only initialized once

**Review by c25vdw — APPROVED** [review]
lgtm

**Pull request merged**